plugins/inputs/logfile/logfile_test.go

```diff
@@ -738,42 +740,44 @@ func TestLogsFileRecreate(t *testing.T) {
 		t.Fatalf("%v log src was returned when 1 should be available", len(lsrcs))
 	}
 
+
 	lsrc := lsrcs[0]
+	defer lsrc.Stop()
 	evts := make(chan logs.LogEvent)
 	lsrc.SetOutput(func(e logs.LogEvent) {
 		if e != nil {
 			evts <- e
 		}
 	})
 
-	go func() {
-		time.Sleep(1 * time.Second)
-
-		// recreate file
-		err = os.Remove(tmpfile.Name())
-		require.NoError(t, err)
-		require.NoError(t, tmpfile.Close())
-		time.Sleep(time.Millisecond * 100)
-		tmpfile, err = os.OpenFile(tmpfile.Name(), os.O_RDWR|os.O_CREATE|os.O_EXCL, 0600)
-		require.NoError(t, err)
-
-		_, err = tmpfile.WriteString(logEntryString + "\n")
-		require.NoError(t, err)
-
-	}()
-
+	tlog.Infof("I! Receive an event (aka new content in file), expect only content from the position in the statefile and onward.")
 	e := <-evts
 	if e.Message() != expectedContent {
 		t.Errorf("Wrong log found before file replacement: \n%v\nExpecting:\n%v\n", e.Message(), expectedContent)
 	}
-	defer lsrc.Stop()
 
-	for {
+	tlog.Infof("I! Deleting the monitored temp file, and then creating it again with the same content...")
+	// recreate file
+	require.NoError(t, tmpfile.Close())
+	require.NoError(t, os.Remove(tmpfile.Name()))
+
+	// Important: Wait long enough for Tail object to detect the file has been deleted and act accordingly.
+	// 100 ms seems to be enough on Linux and MacOS, but not for Windows.
+	time.Sleep(time.Second * 1)
```

Comment:
This 1 line change is required to avoid `os.OpenFile()` failing with error `ACCESS_DENIED`.

If desired I could remove all other changed, but line 677 and line 774.